### PR TITLE
proto: commit generated code to repository

### PR DESCRIPTION
**Pull request by SuperFluffy**
Committing the compiled protobufs to the
repository servces two purposes:

1. it is easier to discover the compiled rust types by simply navigating to the files;
2. users who just want to use the rust crates need not install buf and protoc to compile the protobufs to rust on every build.

While it is convention to not commit generated
code to the repository, I believe the two points
above are good arguments why one should do this
in this case. Also, the protobuf definitions are
not expected to change frequently.

In addition, this commit pins the version of the
remote buf plugins to defend against accidental
breakage.

### CODE REVIEW

**Review by noot — COMMENTED** [review]
can you also update the readme to mention buf isn't required to build anymore (only if protos are modified for development purposes)?

File: crates/astria-proto/src/proto/tonic/astria.execution.v1.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+pub struct InitStateRequest {
+}
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct InitStateResponse {
+    #[prost(bytes="vec", tag="1")]
+    pub block_hash: ::prost::alloc::vec::Vec<u8>,
+}
+include!("astria.execution.v1.tonic.rs");
+// @@protoc_insertion_point(module)
```
Comment (steezeburger) [inline crates/astria-proto/src/proto/tonic/astria.execution.v1.rs:39]: newline nit

Comment (SuperFluffy) [reply crates/astria-proto/src/proto/tonic/astria.execution.v1.rs:39]: This is generated code.

**Comment by SuperFluffy** [comment]
@noot 
> can you also update the readme to mention buf isn't required to build anymore (only if protos are modified for development purposes)?

Updated the README, removed the uncommented code (ty for noticing), and also removed the now unnecessary install step of `buf` and `protobuf-compiler` from the dockerfile.

File: crates/astria-proto/src/proto/tonic/astria.execution.v1.rs (hunk 1 above)
Comment (steezeburger) [reply crates/astria-proto/src/proto/tonic/astria.execution.v1.rs:39]: lmao sorry, it's early! 😄

File: crates/astria-proto/README.md [hunk 2]
```diff
@@ -1,6 +1,32 @@
 # Astria Proto
 
-This repo contains all the protobuf packages for Astria. 
+This repo contains all the protobuf packages for Astria. All rust code
+generated from the protobuf files in [`proto/`](`./proto/`) is commited
+to this repository and no extra tools are needed to encode to/decode from
+protobuf.
+
+Only when changing protobuf definitions (which is done by running running
```
Comment (steezeburger) [inline crates/astria-proto/README.md:8]:
```suggestion
Only when changing protobuf definitions (which is done by running
```

File: crates/astria-proto/README.md [hunk 3]
```diff
@@ -10,9 +36,11 @@ This repo contains all the protobuf packages for Astria.
 
 ## Adding a package
 
-* Create a new folder under `proto/astria/{new_package}`
-* write protos in new folder using package name `astria.{new_package}`
-* update `src/lib.rs` to include your new defintions in a module. Module names should reflect proto package names.
+* Create a new folder `proto/astria/<pkg-name>/<version>`;
```
Comment (steezeburger) [inline crates/astria-proto/README.md:39]: Should this line and the line below end in semi-colons? Maybe period

File: crates/astria-proto/README.md [hunk 4]
```diff
@@ -10,9 +36,11 @@ This repo contains all the protobuf packages for Astria.
...
+* write protos in this folder using the convention name `astria.<pkg-name>.<version>`;
+* update `src/proto/mod.rs` to include your new defintions in a module.
+  Follow the conventions of the other submodules;
```
Comment (steezeburger) [inline crates/astria-proto/README.md:42]: Should this semicolon be a period? or a regular colon?

**Review by steezeburger — APPROVED** [review]
Saw a few issues with the README, but looks good overall

**Review by noot — APPROVED** [review]
can you update the readme in the repo root also to remove that buf is required? otherwise looks good!

**Comment by SuperFluffy** [comment]
I noticed that `buf` has an internal protobuf compiler, so we can remove the extra step installing `protoc` in the github actions.

**Pull request merged**